File: Microsoft.Recognizers.Text.Number/Chinese/Extractors/CardinalExtractor.cs
using Microsoft.Recognizers.Text.Number.Extractors;
using System.Collections.Immutable;
using System.Text.RegularExpressions;

namespace Microsoft.Recognizers.Text.Number.Chinese.Extractors
{
    public class CardinalExtractor : BaseNumberExtractor
    {
        internal sealed override ImmutableDictionary<Regex, string> Regexes { get; }
        protected sealed override string ExtractType { get; } = Constants.SYS_NUM_CARDINAL;

        //CardinalExtractor = Int + Double
        public CardinalExtractor(ChineseNumberMode mode = ChineseNumberMode.Default)
        {
            var builder = ImmutableDictionary.CreateBuilder<Regex, string>();

            var intExtractChs = new IntegerExtractor(mode);
            builder.AddRange(intExtractChs.Regexes);

            var douExtractorChs = new DoubleExtractor();
            builder.AddRange(douExtractorChs.Regexes);

            Regexes = builder.ToImmutable();
        }
    }
}

File: Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/chinese/extractors/CardinalExtractor.java
package com.microsoft.recognizers.text.number.chinese.extractors;

import com.microsoft.recognizers.text.number.Constants;
import com.microsoft.recognizers.text.number.chinese.ChineseNumberExtractorMode;
import com.microsoft.recognizers.text.number.extractors.BaseNumberExtractor;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.regex.Pattern;

public class CardinalExtractor extends BaseNumberExtractor {

    private final Map<Pattern, String> regexes;

    @Override
    protected Map<Pattern, String> getRegexes() {
        return this.regexes;
    }

    @Override
    protected String getExtractType() {
        return Constants.SYS_NUM_CARDINAL;
    }

    public CardinalExtractor() {
        this(ChineseNumberExtractorMode.Default);
    }

    public CardinalExtractor(ChineseNumberExtractorMode mode) {
        HashMap<Pattern, String> builder = new HashMap<>();

        IntegerExtractor intExtractChs = new IntegerExtractor(mode);
        builder.putAll(intExtractChs.getRegexes());

        DoubleExtractor douExtractorChs = new DoubleExtractor();
        builder.putAll(douExtractorChs.getRegexes());

        this.regexes = Collections.unmodifiableMap(builder);
    }
}
